pyomo/core/plugins/transform/bounds_tightener.py

```diff
@@ -0,0 +1,61 @@
+from pyomo.core.base.constraint import Constraint
+from pyomo.core.kernel.numvalue import value
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.repn.canonical_repn import generate_canonical_repn
+from pyomo.util.plugin import alias
+
+import textwrap
+
+class BoundsTightener(IsomorphicTransformation):
```

Comment:
We should also consider a more descriptive name: I expect us to eventually have a number of bounds tighteners (FBBT, OBBT, etc.)